src/weppy.ts

```diff
@@ -0,0 +1,469 @@
+/// <reference path="../weppy.d.ts"/>
+
+// these seem not to be defined in window
+interface Window {
+	XMLHttpRequest?: XMLHttpRequest;
+	XDomainRequest?: XMLHttpRequest;
+	JSON?: JSON;
+}
+
+module WeppyImpl {
+
+	var PROTOCOL_VERSION = 3,
+		PATH_DELIMITER = '.',
+		NAMESPACE_DELIMITER = '::',
+		active = false,
+		options:WeppySettings = {
+			"host": '/weppy',
+			"transport": 'url', // 'url' or 'post'
+			"active": true,
+			"sample": 0.01,
+			"aggregationInterval": 1000,
+			"maxInterval": 5000,
+			"decimalPrecision": 3,
+			"page": 'index',
+			"context": {},
+			"debug": false
+		},
+		initTime = +(new Date),
+		queue, aggregationTimeout, maxTimeout, sentPerformanceData,
+		now = () => {
+			return window.performance && window.performance.now ? window.performance.now() : +(new Date);
+		},
+		log:any = () => {
+			if (options.debug)
+				if (typeof options.debug == 'function')
+					options.debug.apply(window, arguments);
+				else
+					window.console && window.console.log && window.console.log.apply
+						? window.console.log.apply(window.console, arguments) : void 0;
+		},
+		logError:any = () => {
+			window.console && window.console.error && window.console.error.apply &&
+				window.console.error.apply(window.console,arguments);
+		};
+
+	function round(num:number, precision = options.decimalPrecision) {
+		return Math.round(num * Math.pow(10, precision)) / Math.pow(10, precision)
+	}
+
+	function buildPath(path:string, subpath:string, glue:string = PATH_DELIMITER) {
+		return path + (path != '' && subpath != '' ? glue : '' ) + subpath;
```

Comment:
wrap the condition in the ternary operator for ease of reading and also use strict operator `!==`